packages/gestalt/src/Badge.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import styles from './Badge.css';
+
+type Props = {|
+  size?: 'sm' | 'md' | 'lg',
+  text: string,
+|};
+
+export default function Badge(props: Props) {
+  const { size = 'sm', text } = props;
+  const cs = cx(styles.Badge, styles[size]);
+
+  return <span className={cs}>{text}</span>;
```

Comment:
Is there a way to achieve this using `Box` and `Text`? I think as we grow our library we should reuse as much as possible so changes to one attribute system-wide, say font-size, are reflected everywhere

Reply:
We can get close but not completely match this design. In most use cases `Badge` font size has to be 1 size smaller that the parent (because of additional visual treatment). So `Badge` `sm` is the same as `Text` `xs`. So we could either have more consistency between `Badge` and `Text` (but not `Heading`)—just not sure how useful that would be.
Also `Box` border-radius is larger than `Badge`, so `Box` is not fully compatible with this component.

Reply:
@btraut that's a good point and it's highlighting a possible problem with this component in general. If we want to match text's line height, then the badge has to be much smaller—and less effective. If we want to keep it bigger, it will be larger than text's line height and will overlap with text above/below.
I don't necessarily think it's a problem though. Depends what we want this component to be. Currently, we don't have any use cases where it's used in a multi-line text block. It's only used in menus and short headers. If moving forward we want it to be used differently, then we might want to use a different visual element. Discussing with PDS!